ingester/ingester.go

```diff
@@ -286,3 +298,35 @@ func (fi *FirehoseIngester) handleRecordCreate(
 
 	return nil
 }
+
+func (fi *FirehoseIngester) handleRecordDelete(
+	ctx context.Context,
+	log *zap.Logger,
+	recordUri string,
+) (err error) {
+	ctx, span := tracer.Start(ctx, "firehose_ingester.handle_record_delete")
+	defer span.End()
+
+	parts := strings.Split(recordUri, "/")
+
+	if len(parts) < 3 {
+		return fmt.Errorf("malformed record uri '%s'", recordUri)
+	}
+
+	typ := parts[3]
```

Comment:
**thought:** I’m not happy with this. And I should probably turn this into a separate function like `uriToType` or `getTypeFromUri`, so it’s clearer what this does.

Reply:
Good idea! But as it turns out... they apparently also split the path. See [this](https://github.com/bluesky-social/indigo/blob/76a638564246b43613fd61a6666f42d545544dc3/labeler/service.go#L160) and [this](https://github.com/bluesky-social/indigo/blob/76a638564246b43613fd61a6666f42d545544dc3/testing/car_did_repro_test.go#L111C41-L111C41). :disappointed: 

But they assume that the path is correct, which we can’t do because our path is not index `0`.

Reply:
I added a helper for it now but not in the `bluesky` package cause my laptop was close to 0% and I need to head out now for a bit. 😅 I'll finish this later!